src/WebJobs.Script/Metrics/HostMetricsProvider.cs

```diff
@@ -96,8 +96,11 @@ private void AddOrUpdateMetricsCache(string key, long value)
             try
             {
                 var functionActivityStatusProvider = _serviceProvider.GetScriptHostServiceOrNull<IFunctionActivityStatusProvider>();
-                var functionActivityStatus = functionActivityStatusProvider.GetStatus();
```

Comment:
Bundling this fix in (discussed/fixed with @liliankasem) - this was resulting in a null reference exception in failure scenarios (but was fine with working apps).